src/luaopenssl/lua_openssl.cc

```diff
@@ -77,9 +72,6 @@ namespace {
 			case SP_TLS:
 				method = TLSv1_2_method();
 				break;
-			case SP_SSL:
```

Comment:
@dboltovskyi I believe you could simply wrap this into `#ifdef OPENSSL_NO_SSL3` from SSL library header
BTW there is also a directives like `TLS1_1_MINIMAL_VERSION` in SDL to check TLS1.1 and TLS1.2 support which might be applicable for ATF:
```
#if OPENSSL_VERSION_NUMBER < TLS1_1_MINIMAL_VERSION
      SDL_LOG_WARN(
          "OpenSSL has no TLSv1.1 with version lower 1.0.1, set TLSv1.0");
      method = is_server ? TLSv1_server_method() : TLSv1_client_method();
#else
      method = is_server ? TLSv1_1_server_method() : TLSv1_1_client_method();
#endif
```

Reply:
Suggested changes:
```
switch ((SecurityProtocols)type) {
			case SP_AUTO:
				method = SSLv23_method();
				break;
			case SP_TLS:
#if OPENSSL_VERSION_NUMBER < TLS1_1_MINIMAL_VERSION			
				method = TLSv1_2_method();
#else
				method = TLSv1_method();
#endif
				break;
#ifdef OPENSSL_NO_SSL3
			case SP_SSL:
				method = SSLv3_method();
				break;
#endif
			case SP_DTLS:
				method = DTLSv1_method();
				break;
			default:
				printf("Error: Can not create SSL context with security protocol %d\n", type);
				return NULL;
		}
```